Sage/carbon
# feat(text-inputs): add new validations option to text input components FE-4285

### Proposed behaviour

This PR adds (opt in) support for the new validation designs in text input components. The steps to add this were: 

- Add a new internal component (`ValidationMessage`) to be reused in other components.
- Add a `validationRedesignOptIn` prop to `CarbonProvider`. This is `false` by default.
- Update text input components to align with the new design when `validationRedesignOptIn` is set to `true`.

The following components now support the new validation designs: 

- Textbox
- Date
- DateRange
- Decimal
- GroupedCharacter
- Number
- Textarea
- Select
- FilterableSelect
- MultiSelect
- NumeralDate

The option to have the new validation designs on `Fieldset` has been completely disabled for now.

Also increase the spacing between the two inputs on `DateRange` to `24px` from `12px`.

Example of new design: 

<img width="275" alt="image" src="https://user-images.githubusercontent.com/14963680/155289668-33be98c5-4c02-4c34-8d3a-be04a96561a4.png">

<img width="275" alt="image" src="https://user-images.githubusercontent.com/14963680/155289685-7cac7ecc-fe52-4924-8f73-192ae45e3422.png">



### Current behaviour

Currently we don't support the new validation design in any of our components.

### Checklist




#### QA


### Additional context



### Testing instructions

See the new Chromatic snapshots for new validation support in each of the components listed above.

To test manually: 

http://localhost:9001/?path=/story/date-range--validations-string-new-design
http://localhost:9001/?path=/story/date-input--validations-string-new-design
http://localhost:9001/?path=/story/decimal-input--validations-string-new-design
http://localhost:9001/?path=/story/groupedcharacter--validations-string-new-design
http://localhost:9001/?path=/story/number-input--validations-string-new-design
http://localhost:9001/?path=/story/numeral-date--validations-string-new-design
http://localhost:9001/?path=/story/select--validations-string-new-design
http://localhost:9001/?path=/story/select-filterable--validations-string-new-design
http://localhost:9001/?path=/story/select-multiselect--validations-string-new-design
http://localhost:9001/?path=/story/textarea--validations-string-new-design
http://localhost:9001/?path=/story/textbox--validations-string-new-design
